src/adapters/MongoStorageAdapter.js

```diff
@@ -113,9 +132,47 @@ const MongoStorageAdapter = function (uuid) {
 			cb(err, { status: "failed" });
 		});
 	};
-};
 
+	/**
+	 * Get the prefix used to filter keys for particular topics and key prefixes.
+	 *
+	 * @param {object} params
+	 * @param {string} params.topic The topic
+	 * @param {string} params.keyPrefix The key prefix (optional).
+	 * @private
+	 */
+	getKeyPreface = function(params) {
+		const keyPrefix = "keyPrefix" in params ? params.keyPrefix : "";
+		const preface = `${this.getUserPreface()}:${params.topic}:${keyPrefix}`;
```

Comment:
```suggestion
		if (params && params.topic){
			return `${this.getUserPreface()}${params.topic}:${keyPrefix}`;
		} else {
			return `${this.getUserPreface()}`;
		}
```
To support optional topic correctly - and also non-greedy username arguments (needs suggestion to `getUserPreface()` below)